src/redfish.c

```diff
@@ -0,0 +1,891 @@
+    else if (strcasecmp("Queries", opt->key) == 0) {
+      ret = redfish_read_queries(opt, &service->queries);
+      service->queries_num = opt->values_num;
+    } else {
+      ERROR(PLUGIN_NAME ": Invalid configuration option \"%s\".", opt->key);
+    }
+
+    if (ret != 0) {
+      ERROR(PLUGIN_NAME ": Something went wrong processing the service named \
+            \"%s\"",
+            service->name);
+      goto free_string_query;
+    }
+  }
+
+  llentry_t *entry = llentry_create(service->name, service);
+  if (entry != NULL)
+    llist_append(ctx->services, entry);
+  else {
+    ERROR(PLUGIN_NAME ": Failed to create list for service name \"%s\"",
+          service->name);
+    goto free_string_query;
+  }
+
+  return 0;
+
+free_string_query:
+  sfree(service->name);
+  sfree(service->host);
+  sfree(service->user);
+  sfree(service->passwd);
+  sfree(service->token);
+  for (int j = 0; j < service->queries_num; j++)
+    sfree(service->queries[j]);
+
+free_service:
+  sfree(service);
+  return -1;
+
+error:
+  ERROR(PLUGIN_NAME ": Failed to allocate memory for service");
+  return -ENOMEM;
+}
+
+static int redfish_config(oconfig_item_t *cfg_item) {
+  int ret = redfish_preconfig();
+
+  if (ret != 0)
+    return ret;
+
+  for (int i = 0; i < cfg_item->children_num; i++) {
+    oconfig_item_t *child = cfg_item->children + i;
+
+    if (strcasecmp("Query", child->key) == 0)
+      ret = redfish_config_query(child, ctx->queries);
+    else if (strcasecmp("Service", child->key) == 0)
+      ret = redfish_config_service(child);
+    else {
+      ERROR(PLUGIN_NAME ": Invalid configuration option \"%s\".", child->key);
+    }
+
+    if (ret != 0) {
+      redfish_cleanup();
+      return ret;
+    }
+  }
+
+  return 0;
+}
+
+static int redfish_validate_config(void) {
+  /* Service validation */
+  for (llentry_t *llserv = llist_head(ctx->services); llserv != NULL;
+       llserv = llserv->next) {
+    redfish_service_t *service = llserv->value;
+    if (service->name == NULL) {
+      ERROR(PLUGIN_NAME ": A service has no name");
+      return -EINVAL;
+    }
+    if (service->host == NULL) {
+      ERROR(PLUGIN_NAME ": Service \"%s\" has no host attribute",
+            service->name);
+      return -EINVAL;
+    }
+    if ((service->user == NULL) ^ (service->passwd == NULL)) {
+      ERROR(PLUGIN_NAME ": Service \"%s\" does not have user and/or password "
+                        "defined",
+            service->name);
+      return -EINVAL;
+    }
+    if (service->user == NULL && service->token == NULL) {
+      ERROR(PLUGIN_NAME ": Service \"%s\" does not have an user/pass or "
+                        "token defined",
+            service->name);
+      return -EINVAL;
+    }
+    if (service->queries_num == 0)
+      WARNING(PLUGIN_NAME ": Service \"%s\" does not have queries",
+              service->name);
+
+    for (int i = 0; i < service->queries_num; i++) {
+      redfish_query_t *query_query;
+      bool found = false;
```

Comment:
Please use standard C bools, i.e. `_Bool`. Also use `0` instead of `false`.

Reply:
I'm surprised you're asking to change it because previously we were switching from `_Bool` to `bool` and changing `0` to `false` :)

Reply:
@octo Is this a valid or it need a fix?